gateway/src/main/kotlin/Event.kt

```diff
@@ -591,7 +591,7 @@ public data class ChannelDelete(val channel: DiscordChannel, override val sequen
 public data class ChannelPinsUpdate(val pins: DiscordPinsUpdateData, override val sequence: Int?) : DispatchEvent()
 
 public data class TypingStart(val data: DiscordTyping, override val sequence: Int?) : DispatchEvent()
-public data class GuildCreate(val guild: DiscordGuild, override val sequence: Int?) : DispatchEvent()
+public data class GuildCreate(val guild: DiscordPartialGuild, override val sequence: Int?) : DispatchEvent()
```

Comment:
I'm not sure if we should use `DiscordPartialGuild` for this. This rather seems to be a union of `DiscordUnavailableGuild` and `DiscordGuild`. We might instead introduce a sealed superclass for these two and decide on whether to deserialize it as a `DiscordGuild` or a `DiscordUnavailableGuild` based on the presence of the `unavailable` field.

Reply:
So something like this:
```kotlin
// supertype for DiscordUnavailableGuild and DiscordGuild
@Serializable(with = DiscordBaseGuild.Serializer::class)
public sealed class DiscordBaseGuild {
    public abstract val id: Snowflake

    internal object Serializer : JsonContentPolymorphicSerializer<DiscordBaseGuild>(DiscordBaseGuild::class) {
        override fun selectDeserializer(element: JsonElement) =
            when (element.jsonObject["unavailable"]?.jsonPrimitive?.boolean) {
                true -> DiscordUnavailableGuild.serializer()
                false, null -> DiscordGuild.serializer()
            }
    }
}
```
`DiscordGuild.unavailable` could then also be removed.

Reply:
This at least works for the available case in the basic test I did (unavailable is hard to reproduce).

Reply:
do we even need a super type :thinking:

Reply:
I mean what should we put in the event class if there is no supertype? We don't have union types.

Reply:
this also supports my suggestion: https://github.com/discord/discord-api-docs/commit/745685b683dbcfbfe7e8818d25ed63d3eb6343f3